### PR TITLE
Added Portal GDS downloader through PreSigned URLs

### PR DESCRIPTION
**Pull request by victorskl**
* Tailored towards HPC Gadi, Spartan
* Preserved file name using curl -o flag or adjust to desire
* Verbose steps with minimal basic commands & bash script

### CODE REVIEW

**Review by skanwal — CHANGES_REQUESTED** [review]
Thanks @victorskl. 

Just a single comment from my side, other it LGTM and works well.

**Review by victorskl — COMMENTED** [review]
Ah. Can't seem to find comment pointer. Would you mind point it again, please. @skanwal

File: docs/examples/gds_dl_gen.sh [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# Then piping them all together would be like:
+#jq -c '.results[] | .id' gds_records.json | xargs -I % curl -s -H "Authorization: Bearer $PORTAL_TOKEN" "$ENDPOINT/%/presign" | jq -r '.signed_url' > gds_records__presigned.txt
+
+# But. We shall do a tad more such that
+# - using id to generate presigned url
+# - then, using file name to pass-in curl output flag
+jq -cr '.results[] | "\(.id) \(.name)"' gds_records.json | while read -r id name; do
+    url=$(curl -s -H "Authorization: Bearer $PORTAL_TOKEN" "$ENDPOINT/$id/presign" | jq -r '.signed_url')
+    echo "curl -o '$name' '$url'"
+    #break
```
Comment (skanwal) [inline docs/examples/gds_dl_gen.sh:83]: I appended the echo command to a file i.e. 

```
echo "curl -o '$name' '$url'" >> gds_downloader.txt
```

**Review by skanwal — CHANGES_REQUESTED** [review]

File: docs/examples/gds_dl_gen.sh (hunk 1 above)
Comment (victorskl) [reply docs/examples/gds_dl_gen.sh:83]: Ah. Okay, I will adapt that.

Comment (victorskl) [reply docs/examples/gds_dl_gen.sh]: Added with https://github.com/umccr/data-portal-apis/commit/e1bb0dbbede5aeb227684ea8b669218bac38ba1a

**Review by skanwal — APPROVED** [review]
LGTM, thanks.

**Pull request merged**